File: VisaD.Application/Applications/Dtos/ApplicationCommitDto.cs
using VisaD.Application.Candidates.Dtos;
using VisaD.Application.Common.Dtos;

namespace VisaD.Application.Applications.Dtos
{
	public class ApplicationCommitDto : CommitDto
	{
		public PartDto<ApplicantDto> ApplicantPart { get; set; }

		public int CandidateCommitId { get; set; }
		public CandidateCommitDto CandidateCommit { get; set; }

		public PartDto<EducationDto> EducationPart { get; set; }
		public PartDto<TrainingDto> TrainingPart { get; set; }
		public PartDto<DiplomaDto> DiplomaPart { get; set; }
		public PartDto<TaxAccountDto> TaxAccountPart { get; set; }
		public PartDto<DocumentDto> DocumentPart { get; set; }
		public PartDto<RepresentativeDto> RepresentativePart { get; set; }
		public PartDto<PreviousApplicationDto> PreviousApplicationPart { get; set; }
		public PartDto<MedicalCertificateDto> MedicalCertificatePart { get; set; }

		public string RegisterNumber { get; set; }
		public int CreatorUserId { get; set; }

		public bool HasResult { get; set; }
		public ApplicationLotResultDto LotResult { get; set; }

		public string ChangeStateDescription { get; set; }

		public bool HasOtherCommits { get; set; }
	}
}

File: WebApplication/src/modules/application/models/application-commit.dto.ts
import { CandidateCommitDto } from 'src/modules/candidate/models/candidate-commit.dto';
import { CommitState } from '../../../infrastructure/enums/commit-state.enum';
import { PartDto } from '../../../infrastructure/models/part.dto';
import { ApplicantDto } from './applicant.dto';
import { ApplicationLotResultDto } from './application-lot-result.dto';
import { DiplomaDto } from './diploma.dto';
import { DocumentDto } from './document.dto';
import { EducationDto } from './education.dto';
import { MedicalCertificateDto } from './medical-certificate.to';
import { PreviousApplicationDto } from './previous-application.dto';
import { RepresentativeDto } from './representative.dto';
import { TaxAccountDto } from './tax-account.dto';
import { TrainingDto } from './training.dto';

export class ApplicationCommitDto {
  id: number;
  lotId: number;
  state: CommitState;

  hasResult: boolean;
  lotResult: ApplicationLotResultDto;

  applicantPart: PartDto<ApplicantDto>;

  candidateCommitId: number;
  candidateCommit: CandidateCommitDto;

  educationPart: PartDto<EducationDto>;
  trainingPart: PartDto<TrainingDto>;
  taxAccountPart: PartDto<TaxAccountDto>;
  documentPart: PartDto<DocumentDto>;
  diplomaPart: PartDto<DiplomaDto>;
  representativePart: PartDto<RepresentativeDto>;
  previousApplicationPart: PartDto<PreviousApplicationDto>;
  medicalCertificatePart: PartDto<MedicalCertificateDto>;

  changeStateDescription: string;
  hasOtherCommits: boolean;
}
